index.js

```diff
@@ -613,6 +615,8 @@ module.exports = class Hypercore extends EventEmitter {
       return this._updateSnapshot()
     }
 
+    const fork = this.core.tree.fork
+    const preupgradeLength = this.core.tree.length
```

Comment:
Both of these should be from the current snapshot

Reply:
Getting them from the current snapshot if snapshotted now.